exchanges/binance/binance_websocket.go

```diff
@@ -163,112 +146,122 @@ func (b *Binance) wsHandleData(respRaw []byte) error {
+						Exchange: b.Name,
+						OrderID:  orderID,
+						Err:      err,
+					}
+				}
+				var oStatus order.Status
+				oStatus, err = stringToOrderStatus(data.Data.CurrentExecutionType)
+				if err != nil {
+					b.Websocket.DataHandler <- order.ClassificationError{
+						Exchange: b.Name,
+						OrderID:  orderID,
+						Err:      err,
+					}
+				}
+				var p currency.Pair
+				var a asset.Item
+				p, a, err = b.GetRequestFormattedPairAndAssetType(data.Data.Symbol)
+				if err != nil {
+					return err
+				}
+				b.Websocket.DataHandler <- &order.Detail{
+					Price:           data.Data.Price,
+					Amount:          data.Data.Quantity,
+					ExecutedAmount:  data.Data.CumulativeFilledQuantity,
+					RemainingAmount: data.Data.Quantity - data.Data.CumulativeFilledQuantity,
+					Exchange:        b.Name,
+					ID:              orderID,
+					Type:            oType,
+					Side:            oSide,
+					Status:          oStatus,
+					AssetType:       a,
+					Date:            time.Unix(0, data.Data.OrderCreationTime*int64(time.Millisecond)),
+					Pair:            p,
+				}
+			case "listStatus":
+				var data wsListStatus
+				err := json.Unmarshal(respRaw, &data)
+				if err != nil {
+					return fmt.Errorf("%v - Could not convert to listStatus structure %s",
+						b.Name,
+						err)
+				}
+				b.Websocket.DataHandler <- data
 			}
-			var p currency.Pair
-			var a asset.Item
-			p, a, err = b.GetRequestFormattedPairAndAssetType(data.Symbol)
-			if err != nil {
-				return err
-			}
-			b.Websocket.DataHandler <- &order.Detail{
-				Price:           data.Price,
-				Amount:          data.Quantity,
-				ExecutedAmount:  data.CumulativeFilledQuantity,
-				RemainingAmount: data.Quantity - data.CumulativeFilledQuantity,
-				Exchange:        b.Name,
-				ID:              orderID,
-				Type:            oType,
-				Side:            oSide,
-				Status:          oStatus,
-				AssetType:       a,
-				Date:            time.Unix(0, data.OrderCreationTime*int64(time.Millisecond)),
-				Pair:            p,
-			}
-		case "listStatus":
-			var data wsListStauts
-			err := json.Unmarshal(respRaw, &data)
-			if err != nil {
-				return fmt.Errorf("%v - Could not convert to listStatus structure %s",
-					b.Name,
-					err)
-			}
-			b.Websocket.DataHandler <- data
-		default:
-			b.Websocket.DataHandler <- wshandler.UnhandledMessageWarning{Message: b.Name + wshandler.UnhandledMessage + string(respRaw)}
-			return nil
 		}
 	}
-	if stream, ok := multiStreamData["stream"].(string); ok {
-		streamType := strings.Split(stream, "@")
+	if wsStream, ok := multiStreamData["stream"].(string); ok {
+		streamType := strings.Split(wsStream, "@")
 		if len(streamType) > 1 {
 			if data, ok := multiStreamData["data"]; ok {
 				rawData, err := json.Marshal(data)
 				if err != nil {
 					return err
 				}
+
+				pairs, err := b.GetEnabledPairs(asset.Spot)
+				if err != nil {
+					return err
+				}
+
+				format, err := b.GetPairFormat(asset.Spot, true)
```

Comment:
Is Binance websocket actually limited to SPOT only?

Reply:
Yeah, for now, margin subs are for auth only. Added this to Trello to expand perp and delivery contracts.